### DISCUSSION TITLE
ZeroNet service for systemd

### DISCUSSION THREAD

**Discussion by d47081** [zeronet-conservancy/zeronet-conservancy#274] [Show and tell]
Maybe already exist in documentation, just for note my working example:

```
[Unit]
Description=zeronet
Wants=network.target
After=network.target

[Service]
Type=simple
User=zeronet
Group=zeronet
WorkingDirectory=/home/zeronet/zeronet-conservancy/
ExecStart=/home/zeronet/zeronet-conservancy/start-venv.sh --silent
StandardOutput=file:/home/zeronet/debug.log
StandardError=file:/home/zeronet/error.log
Restart=on-failure

[Install]
WantedBy=multi-user.target
```

* `systemctl enable zeronet` - startup on reboot
* `systemctl start zeronet`

Just one question: is [start-venv.sh](https://github.com/zeronet-conservancy/zeronet-conservancy/blob/master/start-venv.sh#L11) supports `--silent` argument?

**Comment by caryoscelus** [comment]
> Just one question: is [start-venv.sh](https://github.com/zeronet-conservancy/zeronet-conservancy/blob/master/start-venv.sh#L11) supports --silent argument?

`start-venv` should forward all arguments to actual `zeronet.py`, so yeah

> Maybe already exist in documentation

i don't i've seen it there, but some [opinionated example](https://aur.archlinux.org/cgit/aur.git/tree/zeronet.service?h=zeronet-conservancy) was in [arch package](https://aur.archlinux.org/packages/zeronet-conservancy)

**Comment by d47081** [comment]
In addition,

recently I'm using this construction for maybe all network-requiring services, because `network.target` does not means `online` and some services could not be launched on system startup without this flag (not sure about ZN)

```
[Unit]
After=network-online.target
Wants=network-online.target
```

**Comment by slrslr** [comment]
I have these lines different or additionally added:

```
[Unit]
Wants=network-online.target
After=network-online.target local-fs.target
RequiresMountsFor=/YourMountPoint

[Service]
RestartSec=2min
StartLimitIntervalSec=0
StartLimitBurst=0
TimeoutStopSec=15
LimitNOFILE=100000:200000
```

**Wants** and **After** lines worked
**RequiresMountsFor** waits once possible external drive (on which are ZN data) is mounted.
**TimeoutStopSec** may give some additional time for the ZN to shutdown cleanly above default timeout?
**LimitNOFILE** should avoid reaching open files limit on a ZN instance with many sites (estimate: 2000 sites = 3500 open files during search)
**ExecStart** line may have not worked correctly for me, I have prefixed the path to zeronet .sh script by "/bin/bash "
**StartLimitIntervalSec** and **StartLimitBurst** set to 0 disables limit of a number of attempts to start the service (for example if the mountpoint RequiresMountsFor is not available for some minutes or hours)
